index.js

```diff
@@ -78,6 +78,25 @@ plugin.hookRequire = function (options) {
   });
 };
 
+plugin.hookRunInThisContext = function (options) {
+  var fileMap = {};
+
+  istanbul.hook.unhookRunInThisContext();
+  istanbul.hook.hookRunInThisContext(function (path) {
+    return !!fileMap[path];
+  }, function (code, path) {
+    return fileMap[path];
+  }, options);
+
+  return through(function (file, enc, cb) {
+    // If the file is already required, delete it from the cache otherwise the covered
+    // version will be ignored.
+    delete require.cache[path.resolve(file.path)];
```

Comment:
I submited in [istanbull#465](https://github.com/gotwarlost/istanbul/pull/465) the new hooks to node vm ...
after then I will include all hooks in gulp-istanbul ... probably I will change this function